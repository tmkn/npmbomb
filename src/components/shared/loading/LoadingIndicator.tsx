/** @jsx jsx */
import { jsx, css, keyframes, Interpolation } from "@emotion/core";
import React, { memo } from "react";

import { mq, primaryColor, primaryColorLight } from "../../../css";
import { Center } from "../center/Center";

export const LoadingIndicator: React.FC = memo(() => {
    const bounce = keyframes`
        0%, 100% { 
            transform: scale(0.0);
        }
        50% { 
            transform: scale(1.0);
        }
    `;

    const sharedStyle: Interpolation = {
        width: "100%",
        height: "100%",
        borderRadius: "50%",
        opacity: 0.6,
        position: "absolute",
        top: 0,
        left: 0,
        animation: `${bounce} 2.0s infinite ease-in-out`
    };

    const style = css({
        [mq[0]]: {
            width: "40px",
            height: "40px",
            position: "relative",
            margin: "100px auto",
            "& div:nth-child(odd)": {
                ...sharedStyle,
                backgroundColor: primaryColor
            },
            "& div:nth-child(even)": {
                ...sharedStyle,
                backgroundColor: primaryColorLight,
                animationDelay: "-1s"
            }
        }
    });

    return (
        <Center>
            <div css={style}>
                <div></div>
                <div></div>
            </div>
        </Center>
    );
});
