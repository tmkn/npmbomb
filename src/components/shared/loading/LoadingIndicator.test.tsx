import React from "react";
import { render, fireEvent, waitForElement } from "@testing-library/react";

import { LoadingIndicator } from "./LoadingIndicator";

describe("<LoadingIndicator />", () => {
    test("matches snapshot", () => {
        const { asFragment } = render(<LoadingIndicator />);

        expect(asFragment()).toMatchSnapshot();
    });
});
