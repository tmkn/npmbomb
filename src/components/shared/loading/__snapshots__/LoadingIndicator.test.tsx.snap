// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-8a5j6a"
  >
    <div
      class="css-176esrn"
    >
      <div />
      <div />
    </div>
  </div>
</DocumentFragment>
`;
